code/game/gamemodes/events/space_weather.dm

```diff
@@ -0,0 +1,172 @@
+/datum/storyevent/bluespace_storm
+	id = "bluespace storm"
+	name = "Bluespace storm"
+	weight = 0.8
+	event_type = /datum/event/bluespace_storm
+	parallel = FALSE
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/bluespace_storm
+	startWhen = 1
+	announceWhen = 5
+	endWhen = 350 //these 350 are failsafes in case setup() SOMEHOW fucks up
+
+/datum/event/bluespace_storm/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/bluespace_storm/announce()
+	command_announcement.Announce("The scanners have detected a bluespace storm near the ship. Bluespace distortions are likely to happen while it lasts.", "Bluespace Storm")
+
+/datum/event/bluespace_storm/end()
+	command_announcement.Announce("The bluespace storm has ended.", "Bluespace Storm")
+
+/datum/event/bluespace_storm/tick()
+	if(prob(2))
+		var/area/A = random_ship_area(filter_maintenance = TRUE, filter_critical = TRUE)
+		bluespace_distorsion(A.random_space())
+
+/datum/storyevent/ion_blizzard
+	id = "ion blizzard"
+	name = "Ion blizzard"
+	weight = 0.9
+	event_type = /datum/event/ion_blizzard
+	parallel = FALSE
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/ion_blizzard
+	startWhen = 1
+	announceWhen = 1
+	endWhen = 350
+
+/datum/event/ion_blizzard/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/ion_blizzard/announce()
+	command_announcement.Announce("A severe ion storm has been detected near the ship. Lighting subsystems are currently overloaded and may not work properly.", "Ion Blizzard")
+
+/datum/event/ion_blizzard/end()
+	command_announcement.Announce("The ion blizzard has ended.", "Ion Blizzard")
+
+/datum/event/ion_blizzard/tick()
+	if(prob(2)) //don't check every single light every single tick jesus christ
+		for(var/obj/machinery/light/L in random_ship_area())
+			L.broken()
+			return
+	else if(prob(80))
+		for(var/obj/machinery/light/L in random_ship_area())
+			L.flick_light(rand(2,5))
+			return
```

Comment:
if we don't put `return` here, every light will either flicker or break at once, and we don't want that to happen.

Reply:
I misunderstood your intention here then.
By the way, wouldn't that always break the lights in the same order? In the same area I mean.
Or, rather, a single lighting bulb, as no check exists for light being broken.

Reply:
random_shield_area should always return... random... ship... areas... so it's not the same order, and yes, it would affect only one bulb at once, and since tick() procs around every two seconds, i think it's fine

Reply:
I see that area is random, but lightbulb is not. If `random_ship_area()` gives, for example, bar area four times, then bar would miss only one light, not four.